share/p2p/peers/manager.go

```diff
@@ -148,14 +148,16 @@ func (m *Manager) Start(startCtx context.Context) error {
 	m.cancel = cancel
 
 	validatorFn := m.metrics.validationObserver(m.Validate)
-	err := m.shrexSub.AddValidator(validatorFn)
-	if err != nil {
-		return fmt.Errorf("registering validator: %w", err)
-	}
 
-	err = m.shrexSub.Start(startCtx)
-	if err != nil {
-		return fmt.Errorf("starting shrexsub: %w", err)
+	if m.params.EnableShrexSub {
+		err := m.shrexSub.AddValidator(validatorFn)
+		if err != nil {
+			return fmt.Errorf("registering validator: %w", err)
+		}
+		err = m.shrexSub.Start(startCtx)
+		if err != nil {
+			return fmt.Errorf("starting shrexsub: %w", err)
+		}
 	}
 
 	headerSub, err := m.headerSub.Subscribe()
```

Comment:
let's also not subscribe here then and not listen? The whole point is that bridge node shouldn't care to listen on the sub, no?